src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorageOptions.cs

```diff
@@ -33,6 +35,10 @@ public class AzureTableStorageOptions : AzureStorageOperationOptions
         public bool IndentJson { get; set; }
         public TypeNameHandling? TypeNameHandling { get; set; }
         public Action<JsonSerializerSettings> ConfigureJsonSerializerSettings { get; set; }
+
+        public virtual Func<IServiceProvider, IGrainStateSerializer> GrainStorageSerializerFactory => GetDefaultSerializer;
```

Comment:
This is *probably* enough, but maybe we should also include the state name in the args, and a doc comment to explain the arguments. We could also add the options object, but that would be unnecessary since it can be retrieved using IServiceProvider + name